crates/vm/src/system/memory/online.rs

```diff
@@ -145,7 +145,7 @@ impl TracingMemory {
             let num_cells = paged_vec.bytes_capacity() / cell_size;
 
             // TMP: hardcoding for now
-            if i <= 3 {
+            if i < 3 {
```

Comment:
back to <= ?

Reply:
i think we only need to push three times for spaces 1, 2, 3 since the index 0 is already initialized

Reply:
ah... we should really sort out this `as_offset` shift business later